src/binder_netreg.c

```diff
@@ -572,12 +588,21 @@ binder_netreg_start_network_scan(
         RADIO_REQ_START_NETWORK_SCAN_1_5;
     GBinderWriter writer;
     guint i, nspecs = 0;
-    const BinderNetRegRadioType* radio_types[N_RADIO_TYPES];
+    const BinderNetRegRadioType* radio_types[(iface < RADIO_INTERFACE_1_5) ?
+        N_RADIO_TYPES : N_RADIO_TYPES_1_5];
 
     /* Which modes are supported and enabled */
-    for (i = 0; i < N_RADIO_TYPES; i++) {
-        if (self->techs & binder_netreg_radio_types[i].mode) {
-            radio_types[nspecs++] = binder_netreg_radio_types + i;
+    if (iface <= RADIO_INTERFACE_1_4) {
```

Comment:
There's another `if (iface <= RADIO_INTERFACE_1_4)` below, why not to move this interface specific logic under a one `if`. It was in the shared place because it was assumed to be shared but apparently it's not.